server/src/main/java/org/elasticsearch/cluster/ClusterInfo.java

```diff
@@ -67,6 +67,10 @@ public ClusterInfo(ImmutableOpenMap<String, DiskUsage> leastAvailableSpaceUsage,
         this.routingToDataPath = routingToDataPath;
     }
 
+    protected ClusterInfo(ClusterInfo delegate) {
```

Comment:
This constructor seems somewhat redundant since its only used in one test spot now, can't we just do this in the test?:

```java
        SizeFakingClusterInfo(ClusterInfo delegate) {
            super(delegate.getNodeLeastAvailableDiskUsages(), delegate.getNodeMostAvailableDiskUsages(), delegate.shardSizes, 
                delegate.routingToDataPath);
        }
```

Reply:
Ah you're right I can get hold of `shardSizes` and `routingToDataPath` that way, thanks.